src/systems/wheel_slip/WheelSlip.cc

```diff
@@ -242,9 +245,30 @@ bool WheelSlipPrivate::Load(const EntityComponentManager &_ecm,
 /////////////////////////////////////////////////
 void WheelSlipPrivate::Update(EntityComponentManager &_ecm)
 {
-  for (const auto &linkSurface : this->mapLinkSurfaceParams)
+  for (auto &linkSurface : this->mapLinkSurfaceParams)
   {
-    const auto &params = linkSurface.second;
+    auto &params = linkSurface.second;
+    const auto * wheelSlipCmdComp =
+      _ecm.Component<components::WheelSlipCmd>(linkSurface.first);
+    if (wheelSlipCmdComp) {
+      const auto & wheelSlipCmdParams = wheelSlipCmdComp->Data();
+      bool changed = math::equal(
```

Comment:
Should this condition be negated? Right now it seems to check if the values have **not** changed.

Reply:
yes, I'm not sure how I didn't notice this.
Thanks for the comment!

Reply:
Fixed in https://github.com/ignitionrobotics/ign-gazebo/pull/1241/commits/d2bce8588293a49cc6cade4cc0f29f8014ff772c.
Updated OP with manual testing instructions.

Reverting https://github.com/ignitionrobotics/ign-gazebo/pull/1241/commits/80dad6988ab53c7fcb69d45f95e6e4fdb0e9f4e5 to have more feedback when testing is useful.